frontend/components/Corpus.js

```diff
@@ -0,0 +1,147 @@
+import React, {useEffect, useState} from "react";
+import * as PropTypes from "prop-types";
+// import STYLES from "./Corpus.module.scss";
+import {getCookie} from "../common";
+import {Modal} from "react-bootstrap";
+
+const Corpus = ({id}) => {
+
+    const [corpusData, setCorpusData] = useState({});
+    const [allDocData, setAllDocData] = useState([]);
+    const [containsDoc, setContainsDoc] = useState({});
+    const [loading, setLoading] = useState(true);
+    const [showModal, setShowModal] = useState(false);
+    const handleOpenModal = () => setShowModal(true);
+    const handleCloseModal = () => setShowModal(false);
+
+    useEffect(() => {
+        fetch(`/api/corpus/${id}`)
+            .then(response => response.json())
+            .then(data => {
+                setCorpusData(data);
+                fetch("/api/all_documents")
+                    .then(response => response.json())
+                    .then(docsData => {
+                        setAllDocData(docsData);
+                        docsData.map((doc) => setContainsDoc((values) => ({
+                            ...values,
+                            [doc.id]: data.documents.includes(doc.id)
+                        })));
+                        setLoading(false);
+                    });
+            });
+
+    }, []);
+
+    const handleCheckBoxChange = (event) => {
+        setContainsDoc((values) => ({
+            ...values,
+            [event.target.id]: !containsDoc[event.target.id]
+        }));
+    };
+
+    const updateDocs = (event) => {
+        event.preventDefault();
+        handleCloseModal();
+        const docList = Object.keys(containsDoc).filter((id) => {
+            return containsDoc[id];
+        });
+        const csrftoken = getCookie("csrftoken");
+        const requestOptions = {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+                "X-CSRFToken": csrftoken
+            },
+            body: JSON.stringify({
+                id: corpusData.id,
+                documents: docList
+            })
+        };
+        fetch("/api/update_corpus_docs", requestOptions)
+            .then(response => response.json())
+            .then(data => {
+                setCorpusData(data);
+            });
+    };
+
+    const docsList = () => {
+        return (
+            <>
+                <h5>Documents:</h5>
```

Comment:
I think this header is the wrong level given the rest of the document. I think the only higher header level above this in the DOM tree is an `<h1>`, so this should probably be an `<h2>`. We can always restyle it if we don't want it to default to the `<h2>` styling.